scripts/globals/bcnm.lua

```diff
@@ -637,57 +641,78 @@ function checkSkip(player, bfid)
     local copStat   = player:getVar("PromathiaStatus")
     local toauStat  = player:getVar("AhtUrganStatus")
     local sofStat   = player:getQuestStatus(JEUNO, dsp.quest.id.jeuno.STORMS_OF_FATE)
-    local mission2_3a = (basty > dsp.mission.id.bastok.THE_EMISSARY_SANDORIA2 or windy > dsp.mission.id.windurst.THE_THREE_KINGDOMS_SANDORIA2) or ((basty == dsp.mission.id.bastok.THE_EMISSARY_SANDORIA2 or windy == dsp.mission.id.windurst.THE_THREE_KINGDOMS_SANDORIA2) and natStat > 9)
-    local mission2_3b = (sandy > dsp.mission.id.sandoria.JOURNEY_TO_BASTOK2 or windy > dsp.mission.id.windurst.THE_THREE_KINGDOMS_BASTOK2) or ((sandy == dsp.mission.id.sandoria.JOURNEY_TO_BASTOK2 or windy == dsp.mission.id.windurst.THE_THREE_KINGDOMS_BASTOK2) and natStat > 10)
-    local mission2_3c = (sandy > dsp.mission.id.sandoria.JOURNEY_TO_WINDURST2 or basty > dsp.mission.id.bastok.THE_EMISSARY_WINDURST2) or ((sandy == dsp.mission.id.sandoria.JOURNEY_TO_WINDURST2 or basty == dsp.mission.id.bastok.THE_EMISSARY_WINDURST2) and natStat > 8)
+    local mission2_3a =
+        player:hasCompletedMission(BASTOK, mi.bastok.THE_EMISSARY_SANDORIA2) or
```

Comment:
This should be revisited in the future. when a single variable (let alone several) is so wide as to require line breaks like this, we're not doing things optimally.

Reply:
our variable names are also getting to be migraine inducing, and the root cause of that is from trying to shorten shit that is already to big.

Reply:
Variable length casts light on problem, but isn't the root cause, in my opinion.

Our checks need to evolve..!
```lua
hasCompletedBCNM(player, dsp.bcnm.NAME) or
dsp.mission.windurst.JOURNEY_TO_WINDURST2.getStage(player) > 8 or  
dsp.mission.bastok.THE_EMISSARY_WINDURST2.getStage(player) > 8  
```
Add shorthands as needed. `miBas` or whatever.

Another alternative is assigning associated missions and stages with mission BCNMs (ie: moar tables!), and then invoking some `checkEntry` function (hugging lead braces for fewer comment lines):

```lua
-- Table of BCNMs
dsp.bcnm = {
    NAME = {
        [dsp.nation.WINDURST] = {
            id = dsp.mission.id.windurst.JOURNEY_TO_WINDURST2,
            stage = 8
        },
        [dsp.nation.BASTOK] = {
            id = dsp.mission.id.bastok.THE_EMISSARY_WINDURST2,
            stage = 8
        }
    }
}

-- Valid entry? check function ideas
function checkEntryIdeaOne(player, bcnm)
    if hasCompletedBCNM(player, bcnm) then return true end
    local nation = player:getNation()
    if bcnm[nation] and player:getCurrentMission(nation) == bcnm[nation].id then
        return getNationMissionStatus(player, nation) == bcnm[nation].stage
    end
    return false
end

function checkEntryIdeaTwo(player, bcnm)
    if hasCompletedBCNM(player, bcnm) then return true end
    local nation = player:getNation()
    if bcnm[nation] then
        return missions.getMission(nation, bcnm[nation].id).getStage(player) == bcnm[nation].stage
    end
    return false
end

-- BCNM listings
local mission2_3a = checkEntryIdeaOneOrTwo(player, dsp.bcnm.BCNM_NAME)
```
Adjust as needed for the non-nation BCNMs. Maybe a tag in the BCNM's table stating the expansion, and then check a simplified version of the player's current mission and stage.

_That aside_:
In our current code, should we be checking for _rank_ for the "already completed" for these particular BCNMs? Can Windurst Rank 3+ players not help with the BCNM fight in Windurst?

Reply:
No hugs. code got enough love from its mommy when it was little.

Reply:
But code is like cats - being squished makes it feel safe and secure~!

Reply:
:cat2: